src/python/pants/backend/python/subsystems/python_native_code.py

```diff
@@ -38,40 +38,16 @@ class EnvironmentAware(Subsystem.EnvironmentAware):
             advanced=True,
         )
 
-
-@dataclass(frozen=True)
-class PythonNativeCodeEnvironment:
-
-    cpp_flags: tuple[str, ...]
-    ld_flags: tuple[str, ...]
-
-    @property
-    def environment_dict(self) -> Dict[str, str]:
-        return {
-            "CPPFLAGS": safe_shlex_join(self.cpp_flags),
-            "LDFLAGS": safe_shlex_join(self.ld_flags),
-        }
-
-
-@rule
-async def resolve_python_native_code_environment(
-    env_aware: PythonNativeCodeSubsystem.EnvironmentAware,
-) -> PythonNativeCodeEnvironment:
-
-    env_vars = await Get(EnvironmentVars, EnvironmentVarsRequest(("CPPFLAGS", "LDFLAGS")))
-
-    def iter_values(env_var: str, values: Sequence[str]):
-        for value in values:
-            if value == f"<{env_var}>":
-                yield from safe_shlex_split(env_vars.get(env_var, ""))
-            else:
-                yield value
-
-    return PythonNativeCodeEnvironment(
-        cpp_flags=tuple(iter_values("CPPFLAGS", env_aware._cpp_flags)),
-        ld_flags=tuple(iter_values("LDFLAGS", env_aware._ld_flags)),
-    )
-
-
-def rules():
-    return [*collect_rules()]
+        @property
+        def environment_dict(self) -> dict[str, str]:
```

Comment:
```suggestion
        def environment_vars_dict(self) -> dict[str, str]:
```
Maybe?

This somewhat conflates environments with environment variables... while environment variables _can_ be overridden by environments via options, they don't have to be.

Unless we expect all access to environment variables to move to being via `Subsystem`s (maybe...?), this results in two APIs for consuming environement variables.

Reply:
Well, not really -- the specific use case for this is for the cases where the subsystem's options are private, and the public API post-processes based on environment variables. This happens _all the time_, and the `EnvironmentVarsRequest` API is brittle -- you have to request the specific env vars ahead of time, and then pass them into the method that consumes them. This makes those subsystems very difficult to consume correctly. Pre-fetching the envvars makes that entirely non-brittle.

That said, the intention here is that `EnvironmentAware.env_vars` is meant to be used solely for those post-processing methods. Would it help if `env_vars` were called `_env_vars` instead?

Reply:
@stuhood to be clear, `environment_dict` was a property that existed on `PythonNativeCodeSubsystem` before today.

Reply:
> to be clear, environment_dict was a property that existed on PythonNativeCodeSubsystem before today.

Ack. Renaming it would I think make things more clear, even though it's existing

Reply:
Sounds good. I don't feel strongly about it.

Reply:
@Eric-Arellano I'll do that now. Too many things named "environment" 🤣